ts/services/SampleService.ts

```diff
@@ -0,0 +1,221 @@
+            }
+
+            if (config.format === 'json') {
+                logger.info(JSON.stringify(response, null, 2));
+            }
+        } catch (e) {
+            logger.error('There was an error listing the elements: ' + e);
+        }
+    }
+
+    public buildSampleUri(elementId, metricId, config, logger) {
+        let queryUri = `${config.endpoint}`;
+        queryUri += `/ui/elements/${elementId}`;
+        queryUri += `/metrics/${metricId}`;
+        queryUri += `/samples?aggregations=ACTUAL`;
+        if (config.rollup) {
+            queryUri += `&rollup=${config.rollup}`;
+        }
+        if (config.duration) {
+            queryUri += `&duration=${config.duration}`;
+        }
+        if (config.startTime) {
+            queryUri += `&startTime=${config.startTime}`;
+        }
+        if (config.endTime) {
+            queryUri += `&endTime=${config.endTime}`;
+        }
+
+        return queryUri;
+    }
+
+    public async summarizeSamples(config, logger): Promise<void> {
+        try {
+            const queryBody = this.buildElementQuery(config, logger);
+            queryBody.sort = new ElasticSearchSortField('name', 'asc', '_last');
+
+            if (!config.startTime) {
+                config.startTime = moment().startOf('month').format();
+            }
+
+            if (!config.endTime) {
+                config.endTime = moment().format();
+            }
+
+            const metricRequest = this.buildRequest(config, queryBody, 'metrics/elasticsearch/metricQuery');
+            const metricResponse = await request(metricRequest);
+            if (config.excessivelogging) {
+                logger.info('METRIC REQUEST');
+                logger.info(JSON.stringify(metricRequest, null, 2));
+                logger.info('METRIC RESPONSE');
+                logger.info(JSON.stringify(metricResponse, null, 2));
+            }
+            const rows = [];
+
+            for (const el of metricResponse.page.content) {
+                const elementId = el.elementId;
+                const metricId = el.id;
+                const metricFqn = el.fqn;
+
+                // get Element name
+                const elementQuery = new ElasticSearchQuery();
+                elementQuery.elementIds = ElasticSearchQueryList.withItem(elementId);
+                elementQuery.sourceFilter = new ElasticSearchSourceFilter([ 'id', 'name', 'type']);
+                const elementResponse = await request({
+                    auth: {
+                        pass: config.password,
+                        user: config.username
+                    },
+                    body: elementQuery,
+                    json: true,
+                    method: 'POST',
+                    uri: `${config.endpoint}/elements/elasticsearch/elementQuery`
+                });
+                const element = {id: elementResponse.page.content[0].id, name: elementResponse.page.content[0].name, type: elementResponse.page.content[0].type};
```

Comment:
Can you break this inline object definition into multiple lines? I did a double take that it was an object.